synapse/rest/client/devices.py

```diff
@@ -347,14 +356,55 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         return 200, result
 
 
+class DehydratedDeviceEventsServlet(RestServlet):
+    PATTERNS = client_patterns(
+        "/org.matrix.msc3814.v1/dehydrated_device/(?P<device_id>[^/]*)/events$",
+        releases=(),
+    )
+
+    def __init__(self, hs: "HomeServer"):
+        super().__init__()
+        self.message_handler = hs.get_device_message_handler()
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastores().main
+
+    class PostBody(RequestBodyModel):
+        next_batch: Optional[StrictStr]
+
+    @cancellable
```

Comment:
Is this really cancellable? Our [docs](https://matrix-org.github.io/synapse/latest/development/synapse_architecture/cancellation.html) imply that using this for non-GET requests is uncommon / risky; I'm mostly worried about doing something like deleting the device but failing to return to the caller? (But I guess if they disconnected then that would happen anyway?)

Reply:
I don't see any compelling reason to make this cancellable - I've removed it.